src/app/services/app-state.service.ts

```diff
@@ -82,12 +82,12 @@ export class AppStateService {
 
     window.parent.postMessage(message, "*");
 
-    // Modifies the app"s URL
+    // Modifies the app's URL
     const queryParams = window.location.search?.substr(1).split("&").map((paramString: string) => {
 
       const [key, value] = paramString.split("=");
 
-      if (key.toLowerCase() !== "rowid") {
+      if (key && key.toLowerCase() !== "rowid") {
```

Comment:
```suggestion
      if (key?.toLowerCase() !== "rowid") {
```

Reply:
This actually doesn't work in this scenario, because the thing I'm guarding against is specifically an empty string. `"".toLowerCase()` is a valid (if not very meaningful) statement, so optional chaining doesn't prevent the statement from progressing, but `if ("")` resolves to false, so that will prevent the block from executing. I'll add in a comment to this effect, because I can definitely understand the suggestion.

Reply:
Makes sense, in that case a comment would be helpful to prevent future refactors of this condition :)